openverse_catalog/dags/providers/provider_api_scripts/freesound.py

```diff
@@ -8,294 +8,249 @@
 
 Notes:                  https://freesound.org/apiv2/search/text'
                         No rate limit specified.
+                        This script can be run either to ingest the full dataset or
+                        as a dated DAG.
 """
-import copy
 import functools
 import logging
 from datetime import datetime
 
 import requests
 from airflow.models import Variable
+from common import constants
 from common.licenses.licenses import get_license_info
 from common.loader import provider_details as prov
-from common.requester import DelayedRequester
-from common.storage.audio import AudioStore
+from providers.provider_api_scripts.provider_data_ingester import ProviderDataIngester
 from requests.exceptions import ConnectionError, SSLError
 from retry import retry
 
 
-LIMIT = 150
-DELAY = 1  # in seconds
-RETRIES = 3
-HOST = "freesound.org"
-ENDPOINT = f"https://{HOST}/apiv2/search/text"
-PROVIDER = prov.FREESOUND_DEFAULT_PROVIDER
-API_KEY = Variable.get("API_KEY_FREESOUND", default_var="not_set")
-FLAKY_EXCEPTIONS = (SSLError, ConnectionError)
-
-HEADERS = {
-    "Accept": "application/json",
-}
-DEFAULT_QUERY_PARAMS = {
-    "format": "json",
-    "token": API_KEY,
-    "query": "",
-    "page_size": LIMIT,
-    "fields": "id,url,name,tags,description,created,license,type,download,"
-    "filesize,bitrate,bitdepth,duration,samplerate,pack,username,"
-    "num_downloads,avg_rating,num_ratings,geotag,previews",
-}
-
-delayed_requester = DelayedRequester(DELAY)
-audio_store = AudioStore(provider=PROVIDER)
-
-logging.basicConfig(
-    format="%(asctime)s - %(name)s - %(levelname)s:  %(message)s",
-    level=logging.INFO,
-)
 logger = logging.getLogger(__name__)
 
-preview_bitrates = {
-    "preview-hq-mp3": 128000,
-    "preview-lq-mp3": 64000,
-    "preview-hq-ogg": 192000,
-    "preview-lq-ogg": 80000,
-}
-
-
-def main(date="all"):
-    """This script pulls the data for a given date from the Freesound,
-    and writes it into a .TSV file to be eventually read
-    into our DB.
-
-    Required Arguments:
-
-    date:  Date String in the form YYYY-MM-DD.  This is the date for
-           which running the script will pull data.
-    """
-
-    logger.info("Begin: Freesound script")
-    licenses = ["Attribution", "Attribution Noncommercial", "Creative Commons 0"]
-    for license_name in licenses:
-        audio_count = _get_items(license_name, date)
-        logger.info(f"Audios for {license_name} pulled: {audio_count}")
-    logger.info("Terminated!")
-
-
-def _get_query_params(
-    license_name="",
-    page_number=1,
-    default_query_params=None,
-):
-    default_query_params = default_query_params or DEFAULT_QUERY_PARAMS
-    return {
-        **default_query_params,
-        "page": str(page_number),
-        "license": license_name,
+
+class FreesoundDataIngester(ProviderDataIngester):
+    batch_limit = 150
+    RETRIES = 3
```

Comment:
```suggestion
    retries = 3
```
But actually this can be removed, since that's the default number of retries!